src/node/internal/internal_http_outgoing.ts

```diff
@@ -128,30 +142,18 @@ class MessageBuffer {
   }
 
   uncork(): void {
-    this.#corked--;
-    this._flush();
-  }
-
-  _flush(): void {
-    if (this.#corked <= 0) {
-      for (const [index, entry] of this.#data.entries()) {
-        if (!entry.written) {
-          entry.written = true;
-          this.#onWrite?.(index, entry);
-          entry.callback?.call(undefined);
-        }
-      }
+    if (this.#corked > 0) {
+      this.#corked--;
     }
   }
 
   get writableLength(): number {
-    return this.#data.reduce<number>((acc, entry) => {
-      return acc + (entry.written && entry.length ? entry.length : 0);
-    }, 0);
+    // Since we process writes immediately, length is always 0
+    return 0;
```

Comment:
do we not buffer writes if `cork()` is called? If so, that would buffer the writes should be reflected in `writableLength`

In Node.js, for instance:

```js
const w = new stream.Writable({ write(d, e, cb) { cb(); });
w.cork();
w.write('hello');
console.log(w.writableLength()); // 5
w.uncork();
console.log(w.writableLength()); // 0
```

Reply:
Or if this is handled elsewhere and is not relevant here, expanding the comment a bit more to indicate that would be helpful.